### ISSUE TITLE
Use an external home assistant client

### ISSUE THREAD

**Issue by nielstron** [btotharye/mycroft-homeassistant#57]
there already exists an home assistant API tool for python, provided by homeassistant itself:

https://developers.home-assistant.io/docs/en/external_api_rest_python.html

We might want to switch to this one instead of the homebrewn version as this one may be more robust. plus we can then focus on building a haclient together with the homeassistant people (joined forced yay)

**Comment by btotharye** [comment]
What I’m using right now is the native rest API which has more options than this python package and doesn’t require having ha installed either

**Comment by ndarilek** [comment]
First, I just want to say that I appreciate all the work that has gone into this skill. Thanks for that!

I spent several hours yesterday trying to get this working with my single-thermostat Home Assistant setup, and I just couldn't. When I dug into the code, I had a hard time tracing the logic.

So I started a major refactor, including switching to the official Home Assistant remote client module, replacing the intent handlers with decorators, using list comprehensions instead of the custom find_entity function, etc. In a couple hours I had my thermostat controlled.

I'm going to continue with this refactor. If anyone wants to help, I'd happily accept it. My HA setup is currently very simple, but I'm blind and HA's web UI has somewhat significant accessibility issues, so having a voice interface to this that just works is very important to me. So I'd appreciate help from folks with more than one thermostat and some trackable devices. :)

I'll happily submit a PR once this refactor is finished, but my coding style seems a bit different than the way things are now, so no hard feelings if it isn't wanted. :)

https://github.com/ndarilek/mycroft-homeassistant Again, the only intent I changed thus far is the intent to set the thermostat temperature. Device tracking coming next, along with deeper thermostat control and eventually switches.
